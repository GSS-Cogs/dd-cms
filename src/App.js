import React, { Component } from 'react';
import { MainPage } from './components';
export class App extends Component {
  render() {
    return <MainPage />;
  }
}
