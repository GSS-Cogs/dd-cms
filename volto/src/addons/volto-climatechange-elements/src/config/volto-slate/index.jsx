import React from 'react';
import { elements } from './SlateElements';
import TitleBlockView from 'volto-slate/blocks/Title/TitleBlockView';
import TitleBlockEdit from 'volto-slate/blocks/Title/TitleBlockEdit';

export default (config) => {

// https://github.com/eea/volto-slate/blob/master/src/editor/config.jsx
// Customise volto-slate config to include govuk-classes within HTML tags.
  config.settings.slate.elements = {
    ...config.settings.slate.elements,
    ...elements
  };

// https://github.com/eea/volto-slate/blob/master/src/blocks/Title/index.js
// The title of a document has its own special widget which is used for rendering.
// Component shadowing wasnt working for whatever reason, so we make changes here.
// The widget set a class of "firstDocumentHeading" as opposed to inheriting the 
// "govuk-heading-xl" class from <h1> tags.

  const className = 'govuk-heading-xl';
  const formFieldName = 'title';

  config.blocks.blocksConfig.title.view = (props) => (
    <TitleBlockView
      {...props}
      className={className}
      formFieldName={formFieldName}
    />
  );
  config.blocks.blocksConfig.title.edit = (props) => (
    <TitleBlockEdit
      {...props}
      className={className}
      formFieldName={formFieldName}
    />
  );

  return config;
};