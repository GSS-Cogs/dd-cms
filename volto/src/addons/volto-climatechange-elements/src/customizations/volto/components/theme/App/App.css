#field-sparql_query {
  min-height: 300px;
}


