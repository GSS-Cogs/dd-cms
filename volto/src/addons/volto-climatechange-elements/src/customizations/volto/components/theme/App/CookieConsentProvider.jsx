import React, { useContext, createContext, useState } from 'react';
import Cookies from 'js-cookie';

const CookieConsentPreferenceContext = createContext(null);
const CookieConsentContext = createContext(null);
const UpdateCookieConsentContext = createContext(null);

export const useCookieConsentPreferenceSet = () =>
  useContext(CookieConsentPreferenceContext);
export const useCookieConsent = () => useContext(CookieConsentContext);
export const useUpdateCookieConsent = () =>
  useContext(UpdateCookieConsentContext);

const getCookieConsentState = () => {
  let cookiesPolicyString = Cookies.get('cookies_policy');
  try {
    let cookiesPolicy = JSON.parse(cookiesPolicyString);
    return cookiesPolicy;
  } catch (e) {
    const defaultConsent = {
      usage: false,
      campaigns: false,
      settings: false,
    };
    Cookies.set('cookies_policy', JSON.stringify(defaultConsent), {
      expires: 365,
      secure: true,
    });
    return defaultConsent;
  }
};

export const CookieConsentProvider = ({ children }) => {
  const [cookiePreferenceSet, setCookiePreferenceSet] = useState(() => {
    const preference = Cookies.get('cookies_preferences_set') === 'true';
    return preference;
  });

  const [cookieConsent, setCookieConsent] = useState(getCookieConsentState);

  const updateConsent = (value) => {
    const consentObject = { usage: value, campaigns: value, settings: value };
    Cookies.set('cookies_preferences_set', 'true', {
      expires: 365,
      secure: true,
    });
    Cookies.set('cookies_policy', JSON.stringify(consentObject), {
      expires: 365,
      secure: true,
    });
    setCookieConsent(consentObject);
    setCookiePreferenceSet(true);
  };

  return (
    <CookieConsentPreferenceContext.Provider value={cookiePreferenceSet}>
      <CookieConsentContext.Provider value={cookieConsent}>
        <UpdateCookieConsentContext.Provider value={updateConsent}>
          {children}
        </UpdateCookieConsentContext.Provider>
      </CookieConsentContext.Provider>
    </CookieConsentPreferenceContext.Provider>
  );
};
