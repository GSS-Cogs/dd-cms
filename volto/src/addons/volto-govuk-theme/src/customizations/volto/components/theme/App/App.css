main {
  display: block;
}

/* big landscape tablets, laptops, and desktops */
@media (min-width: 1025px) {

  .content-area > .volto-width-container--wide {
    min-height: 100%;
    display: grid;
    grid-template-columns: 256px auto;
    column-gap: 32px;
  }

}

.volto-width-container--wide .hamburger {
  padding-left: 0;
  padding-right: 0;
  padding-top: 0;
}

.ui.segment.content-area {
  padding: 0;
  margin: 0;
}

