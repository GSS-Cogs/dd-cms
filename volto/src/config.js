/**
 * Add your config changes here.
 * @module config
 * @example
 * export default function applyConfig(config) {
 *   config.settings = {
 *     ...config.settings,
 *     port: 4300,
 *     listBlockTypes: {
 *       ...config.settings.listBlockTypes,
 *       'my-list-item',
 *    }
 * }
 */

// All your imports required for the config here BEFORE this line
import '@plone/volto/config';
const hotjarMiddleware = require('./hotjar-middleware');

export default function applyConfig(config) {
  if (__SERVER__) {
    const express = require('express');
    const health = express.Router();
    health.all('/_healthcheck', function (req, res, next) {
      return res.status(200).end();
    });
    const auth = express.Router();
    auth.id = 'basic-auth'
    auth.all('*', function (req, res, next) {
      if (process.env['BASIC_AUTH']) { // only do basic auth when this env var is set
        const req_auth = req.get('authorization');
        if (!(typeof (req_auth) === 'string') || // no auth header
          // if there is an authorization header, then
          (
            req.get('authorization').startsWith('Basic ') && // if the header starts with Basic
            !(req.get('authorization') === 'Basic ' + process.env['BASIC_AUTH']) // and it's not the right login
          )
        ) {
          return res
            .status(401)
            .set('WWW-Authenticate', 'Basic realm="Access to beta"')
            .send('Authentication required.');
        }
      }
      return next();
    });
    config.settings.expressMiddleware = [...config.settings.expressMiddleware, hotjarMiddleware, health, auth];
  }
  config.settings.hasWorkingCopySupport = true;
  return config;
}
