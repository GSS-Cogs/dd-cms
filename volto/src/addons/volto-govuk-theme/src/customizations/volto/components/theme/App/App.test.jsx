import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';
import config from '@plone/volto/registry';

import { __test__ as App } from './App';

beforeAll(() => {
  config.settings.navDepth = 1;
  config.views.errorViews = {
    ECONNREFUSED: () => <div className="ECONNREFUSED" />,
  };
});

const mockStore = configureStore();

jest.mock('@plone/volto/components', () => ({
  'Toolbar': () => null,
  'Header': () => null,
  'Breadcrumbs': () => null,
  'Messages': () => null,
  'Navigation': () => null,
  'Footer': () => null,
  'SkipLinks': () => null,
  'OutdatedBrowser': () => null,
  'Icon': () => null,
  'AppExtras': () => null,
}));

jest.mock('semantic-ui-react', () => ({
  Segment: jest.fn(() => <div id="segment" />),
  Container: jest.fn(() => <div id="container" />),
}));

describe('App', () => {
  it('renders a app component', () => {
    const store = mockStore({
      userSession: {
        token: 'abcdefgh',
      },
      content: { data: { id: 'content', '@type': 'Document' } },
      apierror: {},
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/blog/edit']}>
          <App
            location={{ pathname: '/blog/edit' }}
            route={{
              routes: '',
            }}
          >
            <div />
          </App>
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
