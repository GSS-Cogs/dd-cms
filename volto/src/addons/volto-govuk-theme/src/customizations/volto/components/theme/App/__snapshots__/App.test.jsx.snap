// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a app component 1`] = `
Array [
  <div
    id="segment"
  />,
  <div
    className="Toastify"
  />,
]
`;
